homeassistant/components/honeywell/__init__.py

```diff
@@ -47,10 +50,19 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
 
     username = config_entry.data[CONF_USERNAME]
     password = config_entry.data[CONF_PASSWORD]
-
-    client = aiosomecomfort.AIOSomeComfort(
-        username, password, session=async_get_clientsession(hass)
-    )
+    session = async_get_clientsession(hass)
+
+    entries = hass.config_entries.async_entries(DOMAIN)
+    if len(entries) > 1:
+        count = 0
+        for entry in entries:
+            if entry.entry_id == config_entry.entry_id:
+                break
+            count += 1
+        if count > 0:
+            session = async_create_clientsession(hass)
```

Comment:
```suggestion
    if len(entries) > 1:
        session = async_create_clientsession(hass)
    else:
        session = async_get_clientsession(hass)
```
Could we not just do this if there is more than one as you can't really guarantee order of entries as you don't sort it.
Besides maybe this is a bit exceptional with more than one?

Reply:
So make a new session for all entries if more than 1...  I thought the goal was to use the main session whenever possible.